diff -Naur linux.org/sound/soc/codecs/pcm512x.c linux.new/sound/soc/codecs/pcm512x.c
--- linux.org/sound/soc/codecs/pcm512x.c	2018-03-16 17:34:49.000000000 +0100
+++ linux.new/sound/soc/codecs/pcm512x.c	2018-03-23 23:26:36.040574791 +0100
@@ -478,7 +478,7 @@
 
 static const u32 pcm512x_dai_rates[] = {
 	8000, 11025, 16000, 22050, 32000, 44100, 48000, 64000,
-	88200, 96000, 176400, 192000, 384000,
+	88200, 96000, 176400, 192000, 352800, 384000,
 };
 
 static const struct snd_pcm_hw_constraint_list constraints_slave = {
