diff -Naur linux.org/sound/soc/codecs/pcm5102a.c linux.new/sound/soc/codecs/pcm5102a.c
--- linux.org/sound/soc/codecs/pcm5102a.c	2018-03-16 17:34:49.000000000 +0100
+++ linux.new/sound/soc/codecs/pcm5102a.c	2018-03-23 23:50:17.706828077 +0100
@@ -25,7 +25,7 @@
 	.playback = {
 		.channels_min = 2,
 		.channels_max = 2,
-		.rates = SNDRV_PCM_RATE_8000_192000,
+		.rates = SNDRV_PCM_RATE_8000_384000,
 		.formats = SNDRV_PCM_FMTBIT_S16_LE |
 			   SNDRV_PCM_FMTBIT_S24_LE |
 			   SNDRV_PCM_FMTBIT_S32_LE
