diff --color -Naur linux.org/sound/core/pcm_native.c linux.new/sound/core/pcm_native.c
--- linux.org/sound/core/pcm_native.c	2020-07-17 19:08:11.000000000 +0200
+++ linux.new/sound/core/pcm_native.c	2020-07-20 09:05:25.992230369 +0200
@@ -2184,7 +2184,7 @@
 	return snd_interval_refine(hw_param_interval(params, rule->var), &t);
 }
 
-#if SNDRV_PCM_RATE_5512 != 1 << 0 || SNDRV_PCM_RATE_192000 != 1 << 12
+#if SNDRV_PCM_RATE_5512 != 1 << 0 || SNDRV_PCM_RATE_384000 != 1 << 14
 #error "Change this table"
 #endif
 
